### ISSUE TITLE
Maximum number of vehicles in a raid should be as per OG and opened up to modding

### ISSUE THREAD

**Issue by FilmBoy84** [OpenApoc/OpenApoc#999]
As discussed in #998 as to "To Do", so listing here so it's not forgotten. See also the list in #941 

As of PR #998 , the raid mechanism uses the following `int` when specifying the maximal number of vehicles in a raid
`int attack_vehicle_count = 2;`

Skin36 has said that the OG uses the following formula
```
attack_vehicle_count = (bld_count+veh_in_org-1)/bld_count
max(attack_vehicle_count) = 5
```

This should be opened up to modding and the Original Game formula should be implemented (or improved) once Organisations are made to buy/sell vehicles from their garage(s) (also curently a To Do which I'll make a separate PR for)

**Comment by FilmBoy84** [comment]
@Kurtsley Many thanks for #1196

Regarding modding, in the base_gamestate, there is an XML list generated for modding (using the base mod XML otherwise) called `organisation_raid_rules.xml`

This specifies the range of vehicles and conditions under which raid missions will take place in the city and what the vehicles on those missions will attempt to achieve on arrival at their target location (raid, storm, treaty, etc.)

Currently, it does not have an XML tag to define the `max(attack_vehicle_count) = 5` as that is hardcoded. There should be a hook to allow this value to be modified in the XML form

Longer term, we should also look at allowing per-organisation rules for the raid rules (as was the case in some pre-release versions of Apoc) but in the short term, simply allowing this max attack vehicle value to be modified would be useful for modders (I do wonder if we also need to think about a min value too, but this could easily create a host of other issues, so probably best to focus on OG functionality and allow modding of what's there)

**Comment by Kurtsley** [comment]
I see. So would the max number be hardcoded into the xml file instead of the cpp file then? I figure it has to be hardcoded at some point to limit the vehicles to 5.

**Comment by FilmBoy84** [comment]
> I see. So would the max number be hardcoded into the xml file instead of the cpp file then? I figure it has to be hardcoded at some point to limit the vehicles to 5.

Yep, by moving it to the XML to be serialised with the gamestate on load it allows mods to offer an alternative value to 5 by modifying their own version of the base_mod's `organisation_raid_rules.xml`

So, for example, I could modify the max value to 7, 9, 12 whatever (as long as I also ensure enough vehicles are available to each org of course)

Whilst it's not part of `organisation_raid_rules.xml` it's hardcoded in an area not accessible to modding
